### PR TITLE
Tau DQM miniAOD validation for MC

### PR DESCRIPTION
**Pull request by gsaha009**
#### PR description:

Following the last PR #36557, the miniAOD validation is prepared for MC for Run3. This reorganization follows the discussions mentioned below:

- [Proposal of new DQM setup - Aniello Spiezia](https://indico.cern.ch/event/860836/contributions/3628014/attachments/1938401/3213048/AnielloSpiezia_041119.pdf) 
- [Tau DQM Sequence Review - Gage DeZoort & Ece Asilar](https://indico.cern.ch/event/870363/contributions/3670599/attachments/1975330/3287515/Tau_DQM_SequenceReview_Jan21st.pdf)
- [Offline DQM Validation Developments from Tau POG - Ece Asilar](https://indico.cern.ch/event/1186647/contributions/4986214/attachments/2487598/4271494/PPDWorshop.pdf)

Related Twiki page: https://twiki.cern.ch/twiki/bin/viewauth/CMS/SWGuidePFTauIDValidation

#### Latest changes:
 - The configs in `Validation.RecoTau.datatypes` use `ak4GenJets` and `genParticles` like before miniAOD validation to allow the validation steps run successfully on RECO, if necessary.
 - All the requirements for the MiniAOD validation are now cloned into `Validation.RecoTau.RecoTauValidation_cff`
 - The generator level tau decay mode finding algorithm is now modified: will follow discussions and update accordingly! 
 - Switched `StandardValidation` off entirely

#### PR validation:

- PR runs fine in local pipeline
- Tests done with workflow: 

  - 14034.0
  - new tests: 5.1, 135.4, 2018.1, 24034.0, 250205.18, 250408.17, 500205.0 

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for: NA

### CODE REVIEW

**Comment by emanueleusai** [comment]
please test

**Comment by gsaha009** [comment]
@easilar 
Can you please have a look and let me know if you have any ideas!
Thanks :)

**Comment by emanueleusai** [comment]
please test

**Comment by emanueleusai** [comment]
pleass test

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
This is definitely something worth bing inspected:

> * You potentially added 12999 lines to the logs

In fact, in wf 13434.0_TTbar_14TeV+2021FSPU there are a few extra warning already in step2:

```
%MSG-w FilterOnEndPath:  AfterModConstruction 10-Jul-2023 07:41:33 CEST  pre-events
The EDFilter 'TauGenJetDecayModeSelector' with module label 'objectTypeSelectedTauMiniAODValDenominatorModuleZTT' appears on EndPath 'validation_step1'.
The return value of the filter will be ignored.
To suppress this warning, either remove the filter from the endpath,
or explicitly ignore it in the configuration by using cms.ignore().

%MSG
%MSG-w FilterOnEndPath:  AfterModConstruction 10-Jul-2023 07:41:33 CEST  pre-events
The EDFilter 'GenJetSelector' with module label 'kinematicSelectedTauMiniAODValDenominatorZTT' appears on EndPath 'validation_step1'.
The return value of the filter will be ignored.
To suppress this warning, either remove the filter from the endpath,
or explicitly ignore it in the configuration by using cms.ignore().

%MSG
%MSG-w FilterOnEndPath:  AfterModConstruction 10-Jul-2023 07:41:33 CEST  pre-events
The EDFilter 'CandPtrSelector' with module label 'kinematicSelectedTauMiniAODValDenominatorZEE' appears on EndPath 'validation_step1'.
The return value of the filter will be ignored.
To suppress this warning, either remove the filter from the endpath,
or explicitly ignore it in the configuration by using cms.ignore().

%MSG
%MSG-w FilterOnEndPath:  AfterModConstruction 10-Jul-2023 07:41:33 CEST  pre-events
The EDFilter 'CandPtrSelector' with module label 'kinematicSelectedTauMiniAODValDenominatorZMM' appears on EndPath 'validation_step1'.
The return value of the filter will be ignored.
To suppress this warning, either remove the filter from the endpath,
or explicitly ignore it in the configuration by using cms.ignore().

%MSG
%MSG-w FilterOnEndPath:  AfterModConstruction 10-Jul-2023 07:41:33 CEST  pre-events
The EDFilter 'GenJetSelector' with module label 'kinematicSelectedTauMiniAODValDenominatorQCD' appears on EndPath 'validation_step1'.
The return value of the filter will be ignored.
To suppress this warning, either remove the filter from the endpath,
or explicitly ignore it in the configuration by using cms.ignore().
```

But definitely far more in step3, which goes from a size of 1.3 k to 166 k, e.g.

```
%MSG-e endRun:  TauDQMHistEffProducer:efficienciesQCD@endProcessBlock  10-Jul-2023 07:55:33 CEST post-events
 Failed to produce efficiency histogram = RecoTauV/hpsPFTauProducerQCD_hpsPFTauDiscriminationByDecayModeFinding/DecayModeFindingEffpt !!
%MSG
%MSG-e endRun:  TauDQMHistEffProducer:efficienciesQCD@endProcessBlock  10-Jul-2023 07:55:33 CEST post-events
  numerator = RecoTauV/hpsPFTauProducerQCD_hpsPFTauDiscriminationByDecayModeFinding/hpsPFTauDiscriminationByDecayModeFinding_vs_ptTauVisible does not exist.
%MSG
%MSG-e endRun:  TauDQMHistEffProducer:efficienciesQCD@endProcessBlock  10-Jul-2023 07:55:33 CEST post-events
  denominator = RecoTauV/hpsPFTauProducerQCD_ReferenceCollection/nRef_Taus_vs_ptTauVisible does not exist.
%MSG
%MSG-e endRun:  TauDQMHistEffProducer:efficienciesQCD@endProcessBlock  10-Jul-2023 07:55:33 CEST post-events
 Failed to produce efficiency histogram = RecoTauV/hpsPFTauProducerQCD_hpsPFTauDiscriminationByDecayModeFinding/DecayModeFindingEffeta !!
%MSG
%MSG-e endRun:  TauDQMHistEffProducer:efficienciesQCD@endProcessBlock  10-Jul-2023 07:55:33 CEST post-events
  numerator = RecoTauV/hpsPFTauProducerQCD_hpsPFTauDiscriminationByDecayModeFinding/hpsPFTauDiscriminationByDecayModeFinding_vs_etaTauVisible does not exist.
%MSG
%MSG-e endRun:  TauDQMHistEffProducer:efficienciesQCD@endProcessBlock  10-Jul-2023 07:55:33 CEST post-events
  denominator = RecoTauV/hpsPFTauProducerQCD_ReferenceCollection/nRef_Taus_vs_etaTauVisible does not exist.
%MSG
```

and so on so forth,

Please have a look!

File: Validation/RecoTau/interface/TauValidationMiniAOD.h [hunk 1]
```diff
@@ -13,8 +13,11 @@
...
 */
 // Original Author: Aniello Spiezia On August 13, 2019
-// user include files
+// Updated April, 2020 by Ece Asilar and Gage DeZoort
+// Updated July, 2023 by Gourab Saha
 
+// user include files
+//#include "FWCore/Framework/interface/EDAnalyzer.h"
 #include "FWCore/Framework/interface/Event.h"
 #include "FWCore/Framework/interface/MakerMacros.h"
```
Comment (perrotta) [inline Validation/RecoTau/interface/TauValidationMiniAOD.h:22]:
```suggestion
```
Not needed here, the plugin is defined in Validation/RecoTau/src/SealModule.cc

File: Validation/RecoTau/interface/TauValidationMiniAOD.h [hunk 2]
```diff
@@ -25,6 +28,9 @@
 #include "DataFormats/Math/interface/LorentzVector.h"
 #include "DataFormats/TauReco/interface/PFTau.h"
 #include "DataFormats/PatCandidates/interface/Tau.h"
+#include "DataFormats/HepMCCandidate/interface/GenParticle.h"
+#include "DataFormats/JetReco/interface/GenJet.h"
+#include "DataFormats/PatCandidates/interface/PackedGenParticle.h"
 #include "DataFormats/Math/interface/deltaR.h"
```
Comment (perrotta) [inline Validation/RecoTau/interface/TauValidationMiniAOD.h:34]: Not needed here: please move to the configuration file.
A few other include's can be more conveniently moved there, for example MessageLogger (but perhaps also a few more)

File: Validation/RecoTau/src/TauValidationMiniAOD.cc [hunk 3]
```diff
@@ -157,28 +536,205 @@ void TauValidationMiniAOD::analyze(const edm::Event& iEvent, const edm::EventSet
...
+
+      // fill decay mode population plot
+      ntau_vs_dmMap.find("")->second->Fill(taus->size(), matchedTau->decayMode());
+
+      //Fill decay mode migration 2D histogragms
+      //First do a gen Matching
+      unsigned genindex = 0;
+      for (std::vector<reco::GenParticle>::const_iterator genParticle = genParticles->begin();
+           genParticle != genParticles->end();
+           genParticle++) {
```
Comment (perrotta) [inline Validation/RecoTau/src/TauValidationMiniAOD.cc:587]: A range based for loop is more convenient here

File: Validation/RecoTau/src/TauValidationMiniAOD.cc [hunk 4]
```diff
@@ -157,28 +536,205 @@ void TauValidationMiniAOD::analyze(const edm::Event& iEvent, const edm::EventSet
...
+        if (abs(genParticle->pdgId()) == 15) {
+          float gendR = deltaR(matchedTau->eta(), matchedTau->phi(), genParticle->eta(), genParticle->phi());
```
Comment (perrotta) [inline Validation/RecoTau/src/TauValidationMiniAOD.cc:589]: Using `deltaR2` is computationally cheaper (here and elsewhere in this code)

File: Validation/RecoTau/src/TauValidationMiniAOD.cc [hunk 5]
```diff
@@ -157,28 +536,205 @@ void TauValidationMiniAOD::analyze(const edm::Event& iEvent, const edm::EventSet
...
+          int dtrpdgID = std::abs(dtr->pdgId());
+          int dtrstatus = dtr->status();
+          if (dtrpdgID == 12 || dtrpdgID == 14 || dtrpdgID == 16)
+            continue;
+          if (dtrpdgID == 111 || dtrpdgID == 311)
+            nPi0s++;
+          else if (dtrpdgID == 211 || dtrpdgID == 321)
+            nPis++;
+          else if (dtrpdgID == 22) {
+            float dr_taugamma = deltaR(gentau.eta(), gentau.phi(), dtr->eta(), dtr->phi());
```
Comment (perrotta) [inline Validation/RecoTau/src/TauValidationMiniAOD.cc:614]: Using `deltaR2` is computationally cheaper (here and elsewhere in this code)

File: Validation/RecoTau/src/TauValidationMiniAOD.cc [hunk 6]
```diff
@@ -157,28 +536,205 @@ void TauValidationMiniAOD::analyze(const edm::Event& iEvent, const edm::EventSet
...
+              const reco::GenParticle *dtr2 = dynamic_cast<const reco::GenParticle *>(dtr->daughter(idtr2));
+              int dtr2pdgID = std::abs(dtr2->pdgId());
+              if (dtr2pdgID == 12 || dtr2pdgID == 14 || dtr2pdgID == 16)
+                continue;
+              if (dtr2pdgID == 111 || dtr2pdgID == 311)
+                nPi0s++;
+              else if (dtr2pdgID == 211 || dtr2pdgID == 321)
+                nPis++;
+              else if (dtr2pdgID == 22) {
+                float dr_taugamma = deltaR(gentau.eta(), gentau.phi(), dtr2->eta(), dtr2->phi());
```
Comment (perrotta) [inline Validation/RecoTau/src/TauValidationMiniAOD.cc:628]: Using `deltaR2` is computationally cheaper (here and elsewhere in this code)

Comment (gsaha009) [reply Validation/RecoTau/src/TauValidationMiniAOD.cc:628]: Done

File: Validation/RecoTau/src/TauValidationMiniAOD.cc (hunk 5 above)
Comment (gsaha009) [reply Validation/RecoTau/src/TauValidationMiniAOD.cc:614]: Done

File: Validation/RecoTau/src/TauValidationMiniAOD.cc (hunk 4 above)
Comment (gsaha009) [reply Validation/RecoTau/src/TauValidationMiniAOD.cc:589]: Done

File: Validation/RecoTau/src/TauValidationMiniAOD.cc (hunk 3 above)
Comment (gsaha009) [reply Validation/RecoTau/src/TauValidationMiniAOD.cc:587]: Done

File: Validation/RecoTau/interface/TauValidationMiniAOD.h (hunk 1 above)
Comment (gsaha009) [reply Validation/RecoTau/interface/TauValidationMiniAOD.h:22]: Removed

File: Validation/RecoTau/interface/TauValidationMiniAOD.h [hunk 7]
```diff
@@ -13,8 +13,11 @@
 
 */
 // Original Author: Aniello Spiezia On August 13, 2019
-// user include files
+// Updated April, 2020 by Ece Asilar and Gage DeZoort
+// Updated July, 2023 by Gourab Saha
 
+// user include files
+//#include "FWCore/Framework/interface/EDAnalyzer.h"
```
[the remark this answers is not in the document]
Comment (gsaha009) [reply Validation/RecoTau/interface/TauValidationMiniAOD.h:20]: Removed

File: Validation/RecoTau/src/TauValidationMiniAOD.cc [hunk 8]
```diff
@@ -12,143 +12,522 @@
...
+  // List of discriminators and their cuts:
+  discriminators_ = iConfig.getParameter<std::vector<edm::ParameterSet>>("discriminators");
+  // Input primaryVertex collection:
+  edm::InputTag PrimaryVertexCollection_ = edm::InputTag("offlineSlimmedPrimaryVertices");
+  primaryVertexCollectionToken_ = consumes<std::vector<reco::Vertex>>(PrimaryVertexCollection_);
+  // Input genetated particle collection:
+  edm::InputTag prunedGenCollection_ = edm::InputTag("prunedGenParticles");
+  prunedGenToken_ = consumes<std::vector<reco::GenParticle>>(prunedGenCollection_);
+  //packedGenToken_ = consumes<std::vector<pat::PackedGenParticle> >(iConfig.getParameter<edm::InputTag>("packed"));
+  edm::InputTag genJetsCollection_ = edm::InputTag("slimmedGenJets");
```
[the remark this answers is not in the document]
Comment (gsaha009) [reply Validation/RecoTau/src/TauValidationMiniAOD.cc:45]: These tags are now made configurable in `Validation.RecoTau.RecoTauValidationMiniAOD_cfi`

File: Validation/RecoTau/src/TauValidationMiniAOD.cc [hunk 9]
```diff
@@ -12,143 +12,522 @@
...
+  // Information about reference collection:
   extensionName_ = iConfig.getParameter<string>("ExtensionName");
-  discriminators_ = iConfig.getParameter<std::vector<edm::ParameterSet> >("discriminators");
+  // List of discriminators and their cuts:
+  discriminators_ = iConfig.getParameter<std::vector<edm::ParameterSet>>("discriminators");
+  // Input primaryVertex collection:
+  edm::InputTag PrimaryVertexCollection_ = edm::InputTag("offlineSlimmedPrimaryVertices");
+  primaryVertexCollectionToken_ = consumes<std::vector<reco::Vertex>>(PrimaryVertexCollection_);
+  // Input genetated particle collection:
+  edm::InputTag prunedGenCollection_ = edm::InputTag("prunedGenParticles");
```
[the remark this answers is not in the document]
Comment (gsaha009) [reply Validation/RecoTau/src/TauValidationMiniAOD.cc:42]: These tags are now made configurable in Validation.RecoTau.RecoTauValidationMiniAOD_cfi

File: Validation/RecoTau/src/TauValidationMiniAOD.cc [hunk 10]
```diff
@@ -12,143 +12,522 @@
...
-  refCollectionInputTagToken_ = consumes<edm::View<reco::Candidate> >(iConfig.getParameter<InputTag>("RefCollection"));
+  // Input collection to compare to taus:
+  refCollectionInputTagToken_ = consumes<edm::View<reco::Candidate>>(iConfig.getParameter<InputTag>("RefCollection"));
+  // Information about reference collection:
   extensionName_ = iConfig.getParameter<string>("ExtensionName");
-  discriminators_ = iConfig.getParameter<std::vector<edm::ParameterSet> >("discriminators");
+  // List of discriminators and their cuts:
+  discriminators_ = iConfig.getParameter<std::vector<edm::ParameterSet>>("discriminators");
+  // Input primaryVertex collection:
+  edm::InputTag PrimaryVertexCollection_ = edm::InputTag("offlineSlimmedPrimaryVertices");
```
[the remark this answers is not in the document]
Comment (gsaha009) [reply Validation/RecoTau/src/TauValidationMiniAOD.cc:39]: The PrimaryVertexCollection is now configurable

File: Validation/RecoTau/src/TauTagValidation.cc [hunk 11]
```diff
@@ -229,6 +229,8 @@ void TauTagValidation::bookHistograms(DQMStore::IBooker& ibooker,
   int j = 0;
   for (const auto& it : discriminators_) {
     string DiscriminatorLabel = it.getParameter<string>("discriminator");
+    std::cout << "Current discriminator: \n";
+    std::cout << DiscriminatorLabel;
```
[the remark this answers is not in the document]
Comment (gsaha009) [reply Validation/RecoTau/src/TauTagValidation.cc:233]: I commented out these two lines for now

File: Validation/RecoTau/python/RecoTauValidation_cff.py [hunk 12]
```diff
@@ -1,43 +1,90 @@
...
+produceDenominatorMiniAODZMM = cms.Sequence(
+    selectMuonsMiniAOD
+    +selectStableMuonsMiniAOD
+    +kinematicSelectedTauMiniAODValDenominatorZMM
+)
+
+# change to + from *
+selectElectronsMiniAOD = selectElectrons.clone(src=cms.InputTag("prunedGenParticles"))
+selectStableElectronsMiniAOD = selectStableElectrons.clone(src="selectElectronsMiniAOD")
+kinematicSelectedTauMiniAODValDenominatorZEE = kinematicSelectedTauValDenominatorZEE.clone(src=cms.InputTag("selectStableElectronsMiniAOD"))
```
[the remark this answers is not in the document]
Comment (gsaha009) [reply Validation/RecoTau/python/RecoTauValidation_cff.py:32]: Done

File: Validation/RecoTau/python/RecoTauValidation_cff.py [hunk 13]
```diff
@@ -1,43 +1,90 @@
...
+kinematicSelectedTauMiniAODValDenominatorZTT = kinematicSelectedTauValDenominatorZTT.clone(src=cms.InputTag("objectTypeSelectedTauMiniAODValDenominatorModuleZTT"))
+produceDenominatorMiniAODZTT = cms.Sequence(
+    tauGenJetsForMiniAODVal
+    +objectTypeSelectedTauMiniAODValDenominatorModuleZTT
+    +kinematicSelectedTauMiniAODValDenominatorZTT
+)
+
+selectMuonsMiniAOD = selectMuons.clone(src=cms.InputTag("prunedGenParticles"))
+selectStableMuonsMiniAOD = selectStableMuons.clone(src="selectMuonsMiniAOD")
+kinematicSelectedTauMiniAODValDenominatorZMM = kinematicSelectedTauValDenominatorZMM.clone(src=cms.InputTag("selectStableMuonsMiniAOD"))
```
[the remark this answers is not in the document]
Comment (gsaha009) [reply Validation/RecoTau/python/RecoTauValidation_cff.py:22]: Done

File: Validation/RecoTau/python/RecoTauValidation_cff.py [hunk 14]
```diff
@@ -1,43 +1,90 @@
...
+from Validation.RecoTau.dataTypes.ValidateTausOnRealMuonsData_cff import *
+from Validation.RecoTau.dataTypes.ValidateTausOnZEE_cff import *
+from Validation.RecoTau.dataTypes.ValidateTausOnZMM_cff import *
+from Validation.RecoTau.dataTypes.ValidateTausOnZTT_cff import *
+from Validation.RecoTau.dataTypes.ValidateTausOnQCD_cff import *
+
+# Modifying the ref cols for miniAOD
+tauGenJetsForMiniAODVal = tauGenJetsForVal.clone(GenParticles = 'prunedGenParticles')
+objectTypeSelectedTauMiniAODValDenominatorModuleZTT = objectTypeSelectedTauValDenominatorModuleZTT.clone(src=cms.InputTag("tauGenJetsForMiniAODVal"))
+kinematicSelectedTauMiniAODValDenominatorZTT = kinematicSelectedTauValDenominatorZTT.clone(src=cms.InputTag("objectTypeSelectedTauMiniAODValDenominatorModuleZTT"))
```
[the remark this answers is not in the document]
Comment (gsaha009) [reply Validation/RecoTau/python/RecoTauValidation_cff.py:13]: Done

File: Validation/RecoTau/python/RecoTauValidation_cff.py [hunk 15]
```diff
@@ -1,43 +1,90 @@
...
+from Validation.RecoTau.dataTypes.ValidateTausOnRealElectronsData_cff import *
+from Validation.RecoTau.dataTypes.ValidateTausOnRealMuonsData_cff import *
+from Validation.RecoTau.dataTypes.ValidateTausOnZEE_cff import *
+from Validation.RecoTau.dataTypes.ValidateTausOnZMM_cff import *
+from Validation.RecoTau.dataTypes.ValidateTausOnZTT_cff import *
+from Validation.RecoTau.dataTypes.ValidateTausOnQCD_cff import *
+
+# Modifying the ref cols for miniAOD
+tauGenJetsForMiniAODVal = tauGenJetsForVal.clone(GenParticles = 'prunedGenParticles')
+objectTypeSelectedTauMiniAODValDenominatorModuleZTT = objectTypeSelectedTauValDenominatorModuleZTT.clone(src=cms.InputTag("tauGenJetsForMiniAODVal"))
```
[the remark this answers is not in the document]
Comment (gsaha009) [reply Validation/RecoTau/python/RecoTauValidation_cff.py:12]: Done

**Comment by gsaha009** [comment]
> This is definitely something worth bing inspected:
> 
> > * You potentially added 12999 lines to the logs
> 
> In fact, in wf 13434.0_TTbar_14TeV+2021FSPU there are a few extra warning already in step2:
> 
> ```
> %MSG-w FilterOnEndPath:  AfterModConstruction 10-Jul-2023 07:41:33 CEST  pre-events
> The EDFilter 'TauGenJetDecayModeSelector' with module label 'objectTypeSelectedTauMiniAODValDenominatorModuleZTT' appears on EndPath 'validation_step1'.
> The return value of the filter will be ignored.
> To suppress this warning, either remove the filter from the endpath,
> or explicitly ignore it in the configuration by using cms.ignore().
> 
> %MSG
> %MSG-w FilterOnEndPath:  AfterModConstruction 10-Jul-2023 07:41:33 CEST  pre-events
> The EDFilter 'GenJetSelector' with module label 'kinematicSelectedTauMiniAODValDenominatorZTT' appears on EndPath 'validation_step1'.
> The return value of the filter will be ignored.
> To suppress this warning, either remove the filter from the endpath,
> or explicitly ignore it in the configuration by using cms.ignore().
> 
> %MSG
> %MSG-w FilterOnEndPath:  AfterModConstruction 10-Jul-2023 07:41:33 CEST  pre-events
> The EDFilter 'CandPtrSelector' with module label 'kinematicSelectedTauMiniAODValDenominatorZEE' appears on EndPath 'validation_step1'.
> The return value of the filter will be ignored.
> To suppress this warning, either remove the filter from the endpath,
> or explicitly ignore it in the configuration by using cms.ignore().
> 
> %MSG
> %MSG-w FilterOnEndPath:  AfterModConstruction 10-Jul-2023 07:41:33 CEST  pre-events
> The EDFilter 'CandPtrSelector' with module label 'kinematicSelectedTauMiniAODValDenominatorZMM' appears on EndPath 'validation_step1'.
> The return value of the filter will be ignored.
> To suppress this warning, either remove the filter from the endpath,
> or explicitly ignore it in the configuration by using cms.ignore().
> 
> %MSG
> %MSG-w FilterOnEndPath:  AfterModConstruction 10-Jul-2023 07:41:33 CEST  pre-events
> The EDFilter 'GenJetSelector' with module label 'kinematicSelectedTauMiniAODValDenominatorQCD' appears on EndPath 'validation_step1'.
> The return value of the filter will be ignored.
> To suppress this warning, either remove the filter from the endpath,
> or explicitly ignore it in the configuration by using cms.ignore().
> ```
> 
> But definitely far more in step3, which goes from a size of 1.3 k to 166 k, e.g.
> 
> ```
> %MSG-e endRun:  TauDQMHistEffProducer:efficienciesQCD@endProcessBlock  10-Jul-2023 07:55:33 CEST post-events
>  Failed to produce efficiency histogram = RecoTauV/hpsPFTauProducerQCD_hpsPFTauDiscriminationByDecayModeFinding/DecayModeFindingEffpt !!
> %MSG
> %MSG-e endRun:  TauDQMHistEffProducer:efficienciesQCD@endProcessBlock  10-Jul-2023 07:55:33 CEST post-events
>   numerator = RecoTauV/hpsPFTauProducerQCD_hpsPFTauDiscriminationByDecayModeFinding/hpsPFTauDiscriminationByDecayModeFinding_vs_ptTauVisible does not exist.
> %MSG
> %MSG-e endRun:  TauDQMHistEffProducer:efficienciesQCD@endProcessBlock  10-Jul-2023 07:55:33 CEST post-events
>   denominator = RecoTauV/hpsPFTauProducerQCD_ReferenceCollection/nRef_Taus_vs_ptTauVisible does not exist.
> %MSG
> %MSG-e endRun:  TauDQMHistEffProducer:efficienciesQCD@endProcessBlock  10-Jul-2023 07:55:33 CEST post-events
>  Failed to produce efficiency histogram = RecoTauV/hpsPFTauProducerQCD_hpsPFTauDiscriminationByDecayModeFinding/DecayModeFindingEffeta !!
> %MSG
> %MSG-e endRun:  TauDQMHistEffProducer:efficienciesQCD@endProcessBlock  10-Jul-2023 07:55:33 CEST post-events
>   numerator = RecoTauV/hpsPFTauProducerQCD_hpsPFTauDiscriminationByDecayModeFinding/hpsPFTauDiscriminationByDecayModeFinding_vs_etaTauVisible does not exist.
> %MSG
> %MSG-e endRun:  TauDQMHistEffProducer:efficienciesQCD@endProcessBlock  10-Jul-2023 07:55:33 CEST post-events
>   denominator = RecoTauV/hpsPFTauProducerQCD_ReferenceCollection/nRef_Taus_vs_etaTauVisible does not exist.
> %MSG
> ```
> 
> and so on so forth,
> 
> Please have a look!

For step2: Now I am using `cms.ignore()` for the filters that return boolean
For step3: The initial idea was to replace the `StandardValidation` completely and that is why, the `pfTauRunDQMValidation` is kept as an empty sequence in `Validation.RecoTau.DQMMCValidation_cfi` and in `DQMSequences_cfi`. So, these extra lines in the step3 log are gone.

File: Validation/RecoTau/interface/TauValidationMiniAOD.h (hunk 2 above)
Comment (gsaha009) [reply Validation/RecoTau/interface/TauValidationMiniAOD.h:34]: Done

**Comment by emanueleusai** [comment]
please test

**Comment by emanueleusai** [comment]
+1

- re-sign

**Pull request merged**